mautic/mautic
# Staging.bugfix default sms transport

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y 
| New feature? | N 
| Automated tests included? | N
| Issues addressed (#s or URLs) | N
| BC breaks? | N
| Deprecations? | N 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This is  a bugfix for 

> We had a issue with mautic when trying to add slooce as sms transport provider, everything was set up correctly and on Configuration -> Text Message Settings -> Default Transport, the field said slooce but it was only displayed in UI but not saved on configuration.

[//]: # ( As applicable: )

#### Testing setup

1. Temporarily change /app/bundles/SmsBundle/Sms/TransportChain.php function getPrimaryTransport to public
2. Insert following into DashboardController's indexAction to test current transport:

```
/** @var TransportChain $chain */
$chain = $this->get('mautic.sms.transport_chain');
var_dump(get_class($chain->getPrimaryTransport())); die();
        
```

#### Steps to test this PR:

1. Disable all SMS transports and verify Mautic configuration ```/index_dev.php/s/config/edit``` section Text messaging. It should contain information about missing transport
2. Enable SMS transport, verify configuration contains the transport as selected and open Dashboard to check that transport is being used 
3. Enable second SMS transport, select it in the configuration and verify on dashboard
4. Disable second transport and verify on dashboard the first one is used again
